modules/services/gpg-agent.nix

```diff
@@ -6,6 +6,13 @@ let
 
   cfg = config.services.gpg-agent;
 
+  homedir = config.programs.gpg.homedir;
+
+  gpgconf = dir: let
+    f = pkgs.runCommand dir { GNUPGHOME="${homedir}"; }
+      "${pkgs.gnupg}/bin/gpgconf --list-dirs ${dir} > $out";
```

Comment:
Are you sure this works with sandboxing enabled? I am getting `/home/jtojnar/.local/share/gnupg/S.gpg-agent`.

Reply:
I've added a test case, but not sure how to properly test this locally. Right now I'm doing
```shell
nix-shell tests -A run.gpg-agent --option sandbox true --pure
```
which fails only in `--pure` mode because `$GNUPGHOME` is set in my env.

Also I cannot reproduce your exact result, as on failure I get the default value `/run/user/1000/gnupg/S.gpg-agent` and not something prefixed by `homedir`.

Reply:
Ah, the test suite only builds the configuration but does not activate it, so relying on environment variables will lead to failure.

Reply:
It seems the sandbox and `gnupg` do not play nicely together. In the sandbox by default there is no `/run/user/0` (as the sandbox must be run as `root), so this code path is taken:

https://github.com/gpg/gnupg/blob/c6702d77d936b3e9d91b34d8fdee9599ab94ee1b/common/homedir.c#L603-L615

https://github.com/gpg/gnupg/blob/c6702d77d936b3e9d91b34d8fdee9599ab94ee1b/common/homedir.c#L737-L743

I.e., the path prefix falls back to `homedir`. Similar for the Travis build here on GitHub, due to this log message during setup:
```log
mount: /run/user not mounted or bad option
```

Your observation appears to be a feature of `gnupg`, not a bug in the new code. I do not see a meaningful way to address this in a test though, since the point of writing it that way was to take whatever `gnupg` makes of `$GNUPGHOME` and just wire it up consistently, so we do not have to hard code any specifics about it.

We would have none of these problems if `gnupg` did not try to be so damn smart, depend on file system state or start its own agent if none is running...

Reply:
My point was that due to this misfeature, the home-manager module will not work on systems with sandboxing enabled (which should be most systems):

```
$ nix-instantiate --eval -E '(with import <nixpkgs> {}; builtins.readFile (runCommand "agent-socket" { GNUPGHOME="/home/jtojnar/.local/share/gnupg"; } "${gnupg}/bin/gpgconf --list-dirs agent-socket > $out"))'
"/home/jtojnar/.local/share/gnupg/S.gpg-agent\n"
```

Reply:
Ah I see... Had forgotten that `home-manager` can be used as a system module. I am shocked by the unexpected depth of this.